extension/clientcredentialsauthextension/extension.go

```diff
@@ -0,0 +1,106 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientcredentialsauthextension
```

Comment:
I know we probably discussed this in the past, but shouldn't this be under a package named `oauth2clientcredentials`, matching the extension name?

Reply:
To be consistent with other extension packages name, WDYT of:
- `oauth2clientcredentialsextension` or 
- `oauth2clientcredentialsauthextension`

Reply:
`oauth2clientcredentialsauthextension` has my vote

Reply:
`oauth2clientcredentialsauthextension` has my vote too, since we previously agreed on using the 'auth' suffix in addition to the 'extension' suffix (e.g., bearertokenauthextension that was previously merged).

Reply:
Thank you very much @gramidt @secat @jpkrohling Fixed the package name to suggested `oauth2clientcredentialsauthextension`.